kernel/aster-nix/src/syscall/dup.rs

```diff
@@ -23,15 +49,24 @@ pub fn sys_dup2(old_fd: FileDesc, new_fd: FileDesc) -> Result<SyscallReturn> {
     log_syscall_entry!(SYS_DUP2);
     debug!("old_fd = {}, new_fd = {}", old_fd, new_fd);
 
-    let current = current!();
-    let mut file_table = current.file_table().lock();
-    let file = file_table.get_file(old_fd)?.clone();
-    if old_fd != new_fd {
-        // The two file descriptors do not share the close-on-exec flag.
-        if let Some(old_file) = file_table.insert_at(new_fd, file, FdFlags::empty()) {
-            // If the file descriptor `new_fd` was previously open, close it silently.
-            let _ = old_file.clean_for_close();
-        }
+    if old_fd == new_fd {
+        let current = current!();
```

Comment:
Pass `current` to `do_dup3` to avoid calling `current!()` repeatedly.

Reply:
I don't think there is repeated `current` here. `dup2` needs an early return if `old_fd == new_fd`.

Reply:
Yes, you are right.